### PR TITLE
PWGFH: Adding Lc in pKpi in the analysis workflow

### PR DESCRIPTION
**Pull request by DelloStritto**
- First version of the Lc pKpi selection
- First version of the Lc pKpi task
- Insertion of the Lc pKpi analysis in the workflow

@ginnocen @vkucera  @nzardosh

### CODE REVIEW

**Review by jgrosseo — APPROVED** [review]

File: Analysis/DataModel/include/Analysis/HFCandidateSelectionTables.h [hunk 1]
```diff
@@ -21,4 +21,13 @@ DECLARE_SOA_COLUMN(IsSelD0bar, isSelD0bar, int);
 DECLARE_SOA_TABLE(HFSelD0Candidate, "AOD", "HFSELD0CAND", hf_selcandidate::IsSelD0, hf_selcandidate::IsSelD0bar);
 } // namespace o2::aod
 
+namespace o2::aod
+{
+namespace hf_selcandidate_lc
```
Comment (vkucera) [inline Analysis/DataModel/include/Analysis/HFCandidateSelectionTables.h:26]: `hf_selcandidate` should be renamed to `hf_selcandidate_d0`.

File: Analysis/DataModel/include/Analysis/HFCandidateSelectionTables.h [hunk 2]
```diff
@@ -21,4 +21,13 @@ DECLARE_SOA_COLUMN(IsSelD0bar, isSelD0bar, int);
...
+{
+DECLARE_SOA_COLUMN(IsSelLc, isSelLc, int);
+} // namespace hf_selcandidate_lc
+DECLARE_SOA_TABLE(HFSelLcCandidate, "AOD", "HFSELLcCAND", hf_selcandidate_lc::IsSelLc);
```
Comment (vkucera) [inline Analysis/DataModel/include/Analysis/HFCandidateSelectionTables.h:30]: The descriptor should be in capitals.

File: Analysis/DataModel/include/Analysis/HFSecondaryVertex.h [hunk 3]
```diff
@@ -280,6 +280,32 @@ auto InvMassDPlus(const T& candidate)
...
+auto ELc(const T& candidate)
+{
+  return candidate.e(RecoDecay::getMassPDG(4122));
+}
+
+template <typename T>
+auto InvMassLc(const T& candidate)
+{
+  return candidate.m(array{RecoDecay::getMassPDG(kProton), RecoDecay::getMassPDG(kKPlus), RecoDecay::getMassPDG(kPiPlus)});
+}
```
Comment (vkucera) [inline Analysis/DataModel/include/Analysis/HFSecondaryVertex.h:308]: I think there should be a second version of this with proton and pion hypotheses swapped.

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+  int getpTBin(T candpT)
+  {
+    double pTBins[npTBins + 1] = {0, 1., 2., 3., 4., 5., 6., 8., 12., 24., 36.};
+    for (int i = 0; i < npTBins; i++) {
+      if (candpT >= pTBins[i] && candpT < pTBins[i + 1]) {
+        return i;
+      }
+    }
+    return -1;
+  }
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx:74]: There was an update of the D0 selector recently. https://github.com/AliceO2Group/AliceO2/pull/4692
Have a look and update this task accordingly, please.

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+
+    } else {
+      return nSigma = 100; //arbitarily large value
+    }
+    if (nSigma < nSigmaCut) {
+      return true;
+    } else {
+      return false;
+    }
+  }
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx:207]: Fix this as in https://github.com/AliceO2Group/AliceO2/pull/4692

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+    } else if (nPDG == 321) {
+      nSigma = track.tofNSigmaPi();
+    } else {
+      return nSigma = 100; //arbitarily large value
+    }
+    if (nSigma < nSigmaCut) {
+      return true;
+    } else {
+      return false;
+    }
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx:233]: Fix this as in https://github.com/AliceO2Group/AliceO2/pull/4692

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+      auto trackNeg1 = hfCandProng3.index1_as<aod::BigTracksPID>(); //negative daughter
+      auto trackPos2 = hfCandProng3.index2_as<aod::BigTracksPID>(); //positive daughter
+
+      statusLc = 0;
+      topolLc = true;
+      pidLc = -1;
+      proton = -1;
+      kMinus = -1;
+      piPlus = -1;
+
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx:304]: Fix this as in https://github.com/AliceO2Group/AliceO2/pull/4692

File: Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+      topolLc = selectionTopolConjugate(hfCandProng3, trackPos1, trackNeg1, trackPos2);
+
+      if (!topolLc) {
+        hfSelLcCandidate(statusLc);
+        continue;
+      }
+
+      proton = selectionPID(trackPos1, 2212);
+      kMinus = selectionPID(trackNeg1, 321);
+      piPlus = selectionPID(trackPos2, 211);
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx:329]: Fix this as in https://github.com/AliceO2Group/AliceO2/pull/4692

File: Analysis/Tasks/PWGHF/taskLc.cxx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  OutputObj<TH1F> hCt{TH1F("hCt", "3-prong candidates;proper lifetime (D^{#pm}) * #it{c} (cm);entries", 120, -20., 100.)};
+  OutputObj<TH1F> hCPA{TH1F("hCPA", "3-prong candidates;cosine of pointing angle;entries", 110, -1.1, 1.1)};
+  OutputObj<TH1F> hEta{TH1F("hEta", "3-prong candidates;candidate #it{#eta};entries", 100, -2., 2.)};
+  //OutputObj<TH1F> hselectionstatus{TH1F("selectionstatus", "3-prong candidates;selection status;entries", 5, -0.5, 4.5)};
+  OutputObj<TH1F> hImpParErr{TH1F("hImpParErr", "3-prong candidates;impact parameter error (cm);entries", 100, -1., 1.)};
+  OutputObj<TH1F> hDecLenErr{TH1F("hDecLenErr", "3-prong candidates;decay length error (cm);entries", 100, 0., 1.)};
+  //OutputObj<TH1F> hdca{TH1F("hdca", "3-prong candidates;prongs DCA to prim. vertex (cm);entries", 100, -1., 1.)};
+  OutputObj<TH1F> hdca2{TH1F("hdca2", "3-prong candidates;prongs DCA to sec. vertex (cm);entries", 100, 0., 1.)};
+
+  Configurable<int> d_selectionFlagLc{"d_selectionFlagLc", 1, "Selection Flag for Lc"};
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/taskLc.cxx:47]: There should be two hypotheses.

**Pull request merged**